src/mixturemodels.jl

```diff
@@ -0,0 +1,24 @@
+function _mixlogpdf1(d::AbstractMixtureModel, x)
+    # using the formula below for numerical stability
+    #
+    # logpdf(d, x) = log(sum_i pri[i] * pdf(cs[i], x))
+    #              = log(sum_i pri[i] * exp(logpdf(cs[i], x)))
+    #              = log(sum_i exp(logpri[i] + logpdf(cs[i], x)))
+
+    K = ncomponents(d)
+    pri = probs(d)
+
+    lp = map(eachindex(pri)) do i
+            if pri[i] > 0.0
+                return logpdf(component(d, i), x) + log(pri[i])
+            end
+        end
```

Comment:
This won't work correctly since `lp[i]` will be `nothing` if `pri[i] <= 0.0`. Maybe better use something like
```suggestion
    indices = findall(!iszero, pri)
    lp = map(indices) do i
        return logpdf(component(d, i), x) + log(pri[i])
    end
```

Reply:
I originally wanted to do a list comprehension like
```julia
lp = [ logpdf(component(d, i), x) + log(pri[i]) for i in eachindex(pri) if pri[i] > 0.0 ]
```
which does not have the `nothing`'s in `lp`.  This some how does not work with Zygote. 

Then I tried to do the samething with `map` and assumed the `nothing`'s would be filtered out. I should have tested it out in a REPL first. Thanks for catching this.